src/skydb.ts

```diff
@@ -70,10 +70,13 @@ export async function setJSON(
     try {
       const publicKey = pki.ed25519.publicKeyFromPrivateKey({ privateKey: privateKeyBuffer });
       entry = await this.registry.getEntry(toHexString(publicKey), dataKey, opts);
-      revision = entry.entry.revision + 1;
+      revision = entry.entry.revision + BigInt(1);
     } catch (err) {
-      revision = 0;
+      revision = BigInt(0);
     }
+  } else {
+    // Assert the input is 64 bits.
+    checkUint64(revision);
```

Comment:
Shouldn't we verify that the revision number is valid, i.e. `old_rev+1`? Or are we doing that somewhere else?

Reply:
If it's provided, then we use it in the call to Sia. If it's not valid, Sia will report an error. Otherwise we would have to do an extra API call just to get the old revision

Reply:
That makes sense.